src/components/icon/icon.spec.js

```diff
@@ -529,13 +529,18 @@ describe('mdIcon service', function() {
       });
     });
 
+    /*
+     * Previous to Angular 1.6, requesting an icon that is not found would throw no errors. After
+     * 1.6, since we do not have a .catch() handler, it now throws a Possibly Unhandled Rejection
+     * error.
+     */
     describe('icon is not found', function() {
       it('should not throw Error', function() {
         expect(function(){
           $mdIcon('notfound');
 
           $httpBackend.flush();
-        }).not.toThrow();
+        }).not.toThrow('Cannot GET notfoundicon.svg');
```

Comment:
We went back and forth on this single test for almost an hour. The issue is that if we `.catch()` the error, then it definitely won't throw anything, which makes the `not.toThrow()` useless. In ng1.3-1.5, it doesn't throw anything, but in 1.6 it will throw a Possibly Unhandled Rejection because we're not handling it anywhere (which is what was designed; we wanted to ignore this particular type of error).

I agree that it is a bit brittle if the error description changes, but I think I would want this test to fail if the error changes so that we know. I'm slightly worried that if something else throws an error (besides the PUR in 1.6) that it will get swallowed, but we have a lot of other tests that should catch that too.

We honestly talked about just removing the test, but I think it's better to have it than to not.

I think all this test is really trying to say is "if we can't find the icon, we shouldn't throw an icon not found error", and that's all we really want to guarantee.

I'm always open to thoughts on how to fix it if you have a moment to dive in and try some code 😄